src/com/owncloud/android/files/services/FileUploader.java

```diff
@@ -939,6 +798,36 @@ public void uploadFile(String uploadKey) {
                     );
                 }
 
+                if (!uploadResult.isSuccess() && uploadResult.getException() != null) {
+
+                    // if failed due to lack of connectivity, schedule an automatic retry
+                    TransferRequester requester = new TransferRequester();
+                    if (requester.shouldScheduleRetry(this, uploadResult.getException())) {
+                        int jobId = mPendingUploads.buildKey(
+                            mCurrentAccount.name,
+                            mCurrentUpload.getRemotePath()
+                        ).hashCode();
+                        requester.scheduleUpload(
+                            this,
+                            jobId,
+                            mCurrentAccount.name,
+                            mCurrentUpload.getRemotePath()
+                        );
+                        uploadResult = new RemoteOperationResult(
+                            ResultCode.NO_NETWORK_CONNECTION
+                        );
+                    } else {
+                        Log_OC.v(
+                            TAG,
+                            String.format(
+                                "Exception in upload, network is OK, no retry scheduled for %1s in %2s",
+                                mCurrentUpload.getRemotePath(),
+                                mCurrentAccount.name
+                            )
+                        );
+                    }
+                }
+
```

Comment:
You could add here an else case and log "Success OR fail without exception for...", just as you've done in FileDownloader